pkg/tournament/tournament.go

```diff
@@ -879,7 +943,7 @@ func StartAllRoundCountdowns(ctx context.Context, ts TournamentStore, id string,
 
 	for division := range t.Divisions {
 		t.IsStarted = true
-		err := StartRoundCountdown(ctx, ts, id, division, round, false, true, false)
+		err = startRoundAfterLockingTournament(ts, t, ctx, division, round, false)
```

Comment:
the above change of send=true to send=false seems unintended
```suggestion
		err = startRoundAfterLockingTournament(ts, t, ctx, division, round, true)
```

Reply:
I was the big confused. After talking to @domino14 it seems that this send boolean is always set to true currently. It probably had a purpose at some point that was lost in a refactor :/.

Reply:
`startRoundAfterLockingTournament` actually has three parts:
1. pre-check
1. call StartRound
1. send event

the "pre-check" part is quite thorough, but `StartAllRoundCountdowns` has its own pre-check, to try to avoid having later divisions error out after starting earlier divisions. they might do (almost) the same thing for now, but if the pre-check is factored out such that `StartAllRoundCountdowns` can call it too (for a minor performance penalty), this would ensure that they are really checking the same things. maybe that way the `t.Started = true` can be moved to just before `ts.Set`.

the "call `StartRound`" part is effectively a one-liner.

the "send event" part (possibly slow especially if the buffered channel is full) should probably be split out, to be called (by both `StartAllRoundCountdowns` and `StartRoundCountdown`) only after `ts.Set` succeeds. if `ts.Set` fails, the in-memory cache is corrupted, but at least the frontend would reflect what's in db. (this has its own problems though, not sure which way makes more sense.)

(the `send` parameter is still not needed)

Reply:
Hmm, I'm not sure I understand what the actionable items are here.